text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+we will need to create a `package.toml` file.
+
+The `package.toml` file will be similar to the one describe in our documentation:
+
+```toml
+[buildpack]
+uri = "examples/my-multiarch-buildpack"
+# OR a .tgz with the previous folder structure
+uri = "my-multiarch-buildpack.tgz"
+```
+
+But in this case, we actually **remove** the [platform](https://buildpacks.io/docs/reference/config/package-config/) section because it will be taken from the `buildpack.toml`.
+
+Packaging a multi-arch buildpack will require the output to be **publish** to a registry or **saved on disk** in OCI layout format.
+
+```bash
+pack buildpack package my-buildpack --config ./package.toml --publish --multi-arch
+# Or
+pack buildpack package my-buildpack.cnb --config ./package.toml --format file --multi-arch
+```
+
+In these cases each `target` entry corresponds to a different buildpack image that is exported into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+### Package a multi-arch Builder
+
+In case of packing a **Builder**, we assume the following premises: 
+
+1. Buildpack authors migrated their `builder.toml` and [remove the stack](https://github.com/buildpacks/pack/issues/1303) concept.
+2. Multi-architecture `build`, `run` images and `buildpacks` are available for baking into the **Builder**.
+
+A sample `builder.toml` file looks like:
+
+```toml
+# Buildpacks to include in builder, these buildpacks MUST be multi-arch and point to Image Index
+[[buildpacks]]
+uri = "<some uri>"
+
+[run]
+# Runtime images - in case of multi-arch images it must point to Image Index
+[[run.images]]
+image = "index.docker.io/paketobuildpacks/run-jammy-tiny:latest"
+
+[build]
+# This image is used at build-time, in case of multi-arch images it must point to Image Index
+image = "docker.io/paketobuildpacks/build-jammy-tiny:0.2.3"
+
+[[targets]]
+os = "linux"
+arch = "amd64"
+
+[[targets]]
+os = "linux"
+arch = "arm64"
+```
+
+As we can see, the proposal is based on the assumption that the `run-image`, `build-image` and `buildpacks` to include 
+in the builder are **multi-arch artifacts**, and we can reach them by reading an 
+[image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+```bash
+pack builder create my-jammy-builder --config ./builder.toml \ 
+      --multi-arch \ 
+      --publish
+```
+
+In this case, because `targets` are specified and `--multi-arch` is being used `pack` will follow the builder creation process for **each provided platform**, 
+pulling the correct (based on os/arch) buildpacks, build and run images and creating different builders images that are 
+exported and combined into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack Package
+
+As a quick summary, our current process to create a buildpack package involves:
```

Comment:
Would anything change for a composite/meta buildpack? I'd assume that you would need to have multi-arch images for every buildpack in the composite buildpack. Otherwise, would it work the same as a regular component buildpack?

Reply:
That's correct, I will add a new example with the composite builpack, but we are assuming you'd need to have multi-arch images for every buildpack in the composite buildpack